model/reliability/query_test.go

```diff
@@ -252,6 +291,144 @@ func TestGetTaskStatsTwoDocuments(t *testing.T) {
+					require := require.New(t)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						docs, err := GetTaskReliabilityScores(filter)
+						require.NoError(err)
+						require.NotEqual(len(docs), 0)
+					})
+				},
+				"Invalid Combination task1": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
+					require := require.New(t)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						filter.StatsFilter.Tasks = []string{task1}
+						filter.StatsFilter.BuildVariants = []string{variant2}
+						filter.StatsFilter.Distros = []string{distro2}
+						docs, err := GetTaskReliabilityScores(filter)
+						require.NoError(err)
+						require.Len(docs, 0)
+					})
+				},
+				"Invalid Combination task2": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
+					require := require.New(t)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						filter.StatsFilter.Tasks = []string{task2}
+						filter.StatsFilter.BuildVariants = []string{variant1}
+						filter.StatsFilter.Distros = []string{distro1}
+						docs, err := GetTaskReliabilityScores(filter)
+						require.NoError(err)
+						require.Len(docs, 0)
+					})
+				},
+				"task1": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
+					require := require.New(t)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						filter.StatsFilter.Tasks = []string{task1}
+						filter.StatsFilter.BuildVariants = []string{variant1, variant2}
+						filter.StatsFilter.Distros = []string{distro1, distro2}
+						docs, err := GetTaskReliabilityScores(filter)
+						require.NoError(err)
+						require.Len(docs, 2)
+						for _, doc := range docs {
+							require.Equal(doc.TaskName, task1)
+							require.NotEqual(doc.BuildVariant, variant2)
+							require.NotEqual(doc.Distro, distro2)
+						}
+					})
+				},
+				"task2": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
+					require := require.New(t)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						filter.StatsFilter.Tasks = []string{task2}
+						filter.StatsFilter.BuildVariants = []string{variant1, variant2}
+						filter.StatsFilter.Distros = []string{distro1, distro2}
+						docs, err := GetTaskReliabilityScores(filter)
+						require.NoError(err)
+						require.Len(docs, 2)
+						for _, doc := range docs {
+							require.Equal(doc.TaskName, task2)
+							require.NotEqual(doc.BuildVariant, variant1)
+							require.NotEqual(doc.Distro, distro1)
+						}
+					})
+				},
+				"MaxQueryLimit": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
+					require := require.New(t)
+					withCancelledContext(ctx, func(ctx context.Context) {
+						task3 := "task3"
+						variantFmt := "variant %04d"
+						distroFmt := "distro %04d"
+
+						require.NoError(insertManyDailyTaskStats(MaxQueryLimit, project, "r1", task3, variantFmt, distroFmt, day1, 10, 5, 1, 1, 1, 2, 10.5))
```

Comment:
Would it make sense to insert MaxQueryLimit + 1 stats to test that the query correctly returns 1000 docs and not 1001?